pims/norpix_reader.py

```diff
@@ -0,0 +1,269 @@
+                  'gamut': self.metadata['gamut']}
+            return Frame(self.process_func(imdata.astype(self._dtype)),
+                         frame_no=i, metadata=md)
+
+    def _read_timestamp(self):
+        """Read a 6-byte timestamp at the current position in the file.
+
+        Returns a floating-point representation in seconds, and a datetime instance.
+        """
+        if self._timestamp_micro:
+            tsecs, tms, tus = self._timestamp_struct.unpack(self._file.read(8))
+            tfloat = tsecs + float(tms) / 1000. + float(tus) / 1.0e6
+        else:
+            tsecs, tms = self._timestamp_struct.unpack(self._file.read(6))
+            tfloat = tsecs + float(tms) / 1000.
+        return tfloat, datetime.datetime.fromtimestamp(tfloat)
+
+    def _get_time(self, i):
+        """Call _read_timestamp() for a given frame."""
+        self._verify_frame_no(i)
+        with FileLocker(self._file_lock):
+            self._file.seek(self._image_offset + self._image_block_size * i
+                            + self._image_bytes)
+            return self._read_timestamp()
+
+    def get_time(self, i):
+        """Return the time of frame i as a datetime instance.
+
+        Times are stored with millisecond precision.
+
+        Calling this function in a different timezone than where the movie
+        was recorded will result in an offset. The .seq format does not
+        store UTC or timezone information.
+        """
+        return self._get_time(self._map_index(i))[1]
+
+    def get_time_float(self, i):
+        """Return the time of frame i as a floating-point number of seconds."""
+        return self._get_time(self._map_index(i))[0]
+
+    def dump_times_float(self):
+        """Return all frame times in file, as an array of floating-point numbers."""
+        return np.array([self._get_time(i)[0] for i in self._all_indices()])
+
+    @property
+    def filename(self):
+        return self._filename
+
+    @property
+    def pixel_type(self):
+        return self._dtype
+
+    @property
+    def width(self):
+        return self.metadata['width']
+
+    @property
+    def height(self):
+        return self.metadata['height']
+
+    @property
+    def frame_shape(self):
+        return (self.metadata['height'], self.metadata['width'])
+
+    @property
+    def frame_rate(self):
+        return self.metadata['suggested_frame_rate']
+
+    def __len__(self):
+        return self._image_count
+
+    def close(self):
+        self._file.close()
+
+    def __repr__(self):
+        return """<Frames>
+Source: {filename}
+Length: {count} frames
+Frame Shape: {w}w x {h}h
+Pixel Datatype: {dtype}""".format(filename=self.filename,
+                                  count=len(self),
+                                  w=self.frame_shape[0],
+                                  h=self.frame_shape[1],
+                                  dtype=self.pixel_type)
+
+
+class FileLocker(object):
```

Comment:
It would be better to not have this in two places, saves the future headache if we want to make this class smarter.

At least add a note that it is duplicated?

If I were doing this I would make a new file to share it, but I don't care enough to put a PR on top of yours to do it.